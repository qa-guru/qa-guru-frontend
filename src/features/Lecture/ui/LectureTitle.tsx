import React from "react";
import { Typography } from "@mui/material";

interface ILectureTitle {
  title: string;
}

const LectureTitle: React.FC<ILectureTitle> = (props) => {
  const { title } = props;

  return <Typography variant="h2">{title}</Typography>;
};

export default LectureTitle;
