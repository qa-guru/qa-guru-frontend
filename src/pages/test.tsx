import { FC } from "react";
import { useParams } from "react-router-dom";

import TestContainer from "features/test/containers/test-container";

const TestPage: FC = () => {
  const { testId, trainingId, lectureId } = useParams<{
    testId: string;
    trainingId: string;
    lectureId: string;
  }>();

  // Проверяем, что все параметры присутствуют
  if (!testId || !trainingId || !lectureId) {
    return <div>Ошибка: отсутствуют необходимые параметры</div>;
  }

  return (
    <TestContainer
      testId={testId}
      trainingId={trainingId}
      lectureId={lectureId}
    />
  );
};

export default TestPage;
