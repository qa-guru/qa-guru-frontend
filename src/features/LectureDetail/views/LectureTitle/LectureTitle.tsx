import React from "react";
import { Typography } from "@mui/material";
import { ILectureTitle } from "./LectureTitle.types";

const LectureTitle: React.FC<ILectureTitle> = (props) => {
  const { title } = props;

  return (
    <Typography mb={3} variant="h4">
      {title}
    </Typography>
  );
};

export default LectureTitle;
