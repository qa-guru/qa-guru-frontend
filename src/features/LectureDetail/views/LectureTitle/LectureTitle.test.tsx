import { MemoryRouter } from "react-router-dom";
import LectureTitle from "./LectureTitle";
import { render } from "../../../../test/utilsTest";

describe("LectureTitle", () => {
  it("the component is render", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <LectureTitle title="Title" />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
