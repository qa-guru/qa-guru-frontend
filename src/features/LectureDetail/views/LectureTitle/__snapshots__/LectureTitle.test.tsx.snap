// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LectureTitle > the component is render 1`] = `
<DocumentFragment>
  <h4
    class="MuiTypography-root MuiTypography-h4 css-1h02dxw-MuiTypography-root"
  >
    Title
  </h4>
</DocumentFragment>
`;
